diagnostics/README.md

```diff
@@ -42,6 +42,16 @@ To get a **full** dump use the following set of flags
 | `NGRAPH_TF_DUMP_GRAPHS=1`    | Dump TF graphs for different passes: precapture, capture, unmarked, marked, clustered, declustered, encapsulated |
 | `TF_CPP_MIN_VLOG_LEVEL=1`    | Enable TF CPP logs                    |
 | `NGRAPH_TF_DUMP_DECLUSTERED_GRAPHS=1` | Dump graphs with final clusters assigned. Use this to view TF computation graph with colored nodes indicating clusters|
+|
+
+### Visualizing encapsulates using TB
+
+* Run your script with this flag: ```NGRAPH_TF_DUMP_DECLUSTERED_GRAPHS=1 python run_TF_network.py```
+* Change directory to this diagnostics folder
+* Run this script to parse the dumped graphs to know which encapsulate a node belongs to. At this step nodemap.pkl is created: ```python get_node_encapsulate_map.py ./path/to/folder/where/run_TF_network.py/exists/where/the/dumps/were/created/in/the/last/step/ nodemap.pkl```
+* Modify the graphdef and dump TB file in ```./vis``` using encapsulate information in ```nodemap.pkl```: ```python ngtf_graph_viewer.py -c nodemap.pkl ./path/to/original_network_pbtxtfile.pbtxt ./vis```. If you do not have the pbtxt of the original tensorflow graph, you can dump it from your script using [write_graph](https://www.tensorflow.org/api_docs/python/tf/train/write_graph)
+*  View the original network with encapsulate information by running tensorboard, using the files created in ```./vis```.
+* Note: you may need to preprocess your protobuffers (pbtxt) to remove the `_class` attribute if any of the above steps result in an error such as `{NodeType} expects to be colocated with unknown node ...`. In that case, first run ```python remove_protobuf_class_attribute.py -d /path/to/pbtxt/to/process/``` and then follow the above steps again.
```

Comment:
Lets move the `Another way of protobuf visualization` section here and add info that in case we have a grappler build we can do it the easy way, or if we dont, we need to do it the hard way (as mentioned in the green parts being re-added here)